cpp/src/arrow/testing/gtest_util.h

```diff
@@ -532,4 +532,9 @@ class ARROW_TESTING_EXPORT GatingTask {
   std::shared_ptr<Impl> impl_;
 };
 
+/// \brief modify an array so that the buffer at index 1 (if it has one) is unaligned
+ARROW_TESTING_EXPORT std::shared_ptr<ArrayData> UnalignValues(const ArrayData& array);
+/// \brief modify an array so that the buffer at index 1 (if it has one) is unaligned
+ARROW_TESTING_EXPORT std::shared_ptr<Array> UnalignValues(const Array& array);
```

Comment:
Why not instead "unalign" all buffers, even if no particular alignment is required? This would probably make things more interesting?

Reply:
I now unalign all buffers.  It was simpler anyways.